src/FS.Embedded/Manifest/ManifestParser.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest
+{
+    internal static class ManifestParser
+    {
+        private static readonly string DefaultManifestName = "Microsoft.Extensions.FileSystem.Embedded.Manifest.xml";
+
+        public static EmbeddedFilesManifest Parse(Assembly assembly)
+        {
+            return Parse(assembly, DefaultManifestName);
+        }
+
+        public static EmbeddedFilesManifest Parse(Assembly assembly, string name)
+        {
+            if (assembly == null)
+            {
+                throw new ArgumentNullException(nameof(assembly));
+            }
+
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            var stream = assembly.GetManifestResourceStream(name);
+            if (stream == null)
+            {
+                throw new InvalidOperationException($"Can't find a manifest with name '{name}'");
+            }
+
+            var document = XDocument.Load(stream);
+
+            var manifest = EnsureElement(document, "Manifest");
+            var manifestVersion = EnsureElement(manifest, "ManifestVersion");
+            var version = EnsureText(manifestVersion);
+            if (!string.Equals("1.0", version))
+            {
+                throw new InvalidOperationException($"Invalid manifest version: '{version}'");
```

Comment:
Add more context to the error message. 
Suggested wording:
```
The embedded file manifest '{manfiestName}' for assembly '{assemblyName}' specifies an unsupported file format version: '{version}'.
```

Also, how would a user go about resolving this?

Reply:
This issue will be cause most likely if we ever update the format of the manifest in a future release in a way that forces us to rev the version.
This will would be caused by a library compiled against a newer version of this package, being consumed by an older version of this assembly.
I'm not sure how to word that succinctly and if it would be valuable to do so.

Do you have any suggestion?